kubernetes/resource_kubernetes_job_test.go

```diff
@@ -102,14 +103,94 @@ func TestAccKubernetesJob_basic(t *testing.T) {
+					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.parallelism", "2"),
+					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.template.0.spec.0.container.0.name", "hello"),
+					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.template.0.spec.0.container.0.image", imageName),
+					resource.TestCheckResourceAttr("kubernetes_job.test", "wait_for_completion", "false"),
+					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.manual_selector", "false"),
+				),
+			},
+			{
+				Config: testAccKubernetesJobConfig_updateMutableFields(name, imageName, "121", "10", "false", "2"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesJobExists("kubernetes_job.test", &conf2),
+					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.active_deadline_seconds", "121"),
+					testAccCheckKubernetesJobForceNew(&conf1, &conf2, false),
+				),
+			},
+			{
+				Config: testAccKubernetesJobConfig_updateMutableFields(name, imageName, "121", "11", "false", "2"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesJobExists("kubernetes_job.test", &conf2),
+					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.backoff_limit", "11"),
+					testAccCheckKubernetesJobForceNew(&conf1, &conf2, false),
+				),
+			},
+			{
+				Config: testAccKubernetesJobConfig_updateMutableFields(name, imageName, "121", "11", "true", "2"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesJobExists("kubernetes_job.test", &conf2),
+					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.manual_selector", "true"),
+					testAccCheckKubernetesJobForceNew(&conf1, &conf2, false),
+				),
+			},
+			{
+				Config: testAccKubernetesJobConfig_updateMutableFields(name, imageName, "121", "11", "true", "3"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesJobExists("kubernetes_job.test", &conf2),
+					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.parallelism", "3"),
+					testAccCheckKubernetesJobForceNew(&conf1, &conf2, false),
+				),
+			},
+			{
+				Config: testAccKubernetesJobConfig_updateImmutableFields(name, imageName, "12"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesJobExists("kubernetes_job.test", &conf2),
+					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.completions", "12"),
+					testAccCheckKubernetesJobForceNew(&conf1, &conf2, true),
+				),
+			},
+			{
+				Config: testAccKubernetesJobConfig_updateImmutableFields(name, imageName1, "12"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesJobExists("kubernetes_job.test", &conf3),
+					resource.TestCheckResourceAttr("kubernetes_job.test", "spec.0.template.0.spec.0.container.0.image", imageName1),
+					testAccCheckKubernetesJobForceNew(&conf2, &conf3, true),
+				),
+			},
+		},
+	})
+}
+
```

Comment:
This whole test is a result of looking at Patrick's work and seeing that certain fields are allowed to change in the Job resource. I tested manually using `kubectl edit` to figure out exactly which fields are mutable vs immutable. The API docs don't always accurately reflect that information.